### PR TITLE
EREGCSC-1722 - New File/location for serverless backend

### PR DESCRIPTION
**Pull request by thwalker6**
Resolves # EREGCSC-1722

**Description-**

This is for a serverless file that can be deployed to any aws environment using serverless.  The folder structure will contain more serverless that we would possibly want to implement on there as well.  

**This pull request changes...**

- Nothing in the app.  Just is a creation of a serverless file that has no dependencies on our environment.

**Steps to manually verify this change...**

1. Spelled out in the documentation on the ticket but if you would like to use it
2. 
Set up serverless to point to the desired environment
serverless config credentials --provider AWS --key YOUR_ACCESS_KEY --secret YOUR_ACCESS_SECRET 
Set up environment variables in AWS AppConfig.	
We use a variety of application configurations that are used as environment variables into the serverless document.  Make sure you add them.  As of right now these are the following needed.  Make sure the file paths match up 
/eregulations/db/password
As more configs are added this list will increase
Set your folder to the cmcs-eregulations/solution/serverless-test-files folder for the project (essentially root for the project
Run the command serverless deploy --config serverless-backend.yml

Open up CloudFormation.  This will be deploying a stack called cmcs-eregs-site.  As it builds the application it will say how it is going with the various resources.  This file builds the database cluster and instance, a role for handling lambda functions, security groups, subnet groups, and subnets.
After the resources are built, check for any failures.  CloudFormation will tell you under events what resources fail to build.  If any fail to build, check out the error messages and address them. The base file should not have any but it is possible to have some issues.
Now go ahead and make your changes you want.  Upgrading the database, changing security groups, etc.  Depending on the resources you change you might have different changes such as deletion, retention, snapshots.  Check AWS documentation on the resources to verify the effects.

### CODE REVIEW

File: solution/serverless-test-files/serverless-backend.yml [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    DB_PASSWORD: ${ssm:/eregulations/db/password}
+    DB_HOST: !GetAtt [RDSResource, Endpoint.Address]
+    DB_PORT: !GetAtt [RDSResource, Endpoint.Port]
+    #SURVEY_URL: ${ssm:/eregulations/survey_url}
+  vpc:
+    securityGroupIds:
+      - !Ref ServerlessSecurityGroup
+    subnetIds:
+      - subnet-021d6d373ec2d8c86
+      - subnet-0121c25b7306fe803
```
Comment (peggles2) [inline solution/serverless-test-files/serverless-backend.yml:23]: I'm getting this error
Serverless: Operation failed!
Serverless: View the full error output: https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aus-east-1%3A666435484074%3Astack%2Fcmcs-eregs-site-dev%2F05d423d0-a665-11ed-934b-0e74c018307b

 Serverless Error ----------------------------------------

  An error occurred: DBSubnetGroup - Resource handler returned message: "Some input subnets in :[subnet-021d6d373ec2d8c86, subnet-0121c25b7306fe803] are invalid. (Service: Rds, Status Code: 400, Request ID: 4245a37d-7cc9-4ef7-9ea8-7e8e51053bae)" (RequestToken: d0d2e84b-c918-109f-5991-e1957db5c04f, HandlerErrorCode: InvalidRequest).

File: solution/serverless-test-files/serverless-backend.yml [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    - node_modules/**
+    - nginx/**
+
+custom:
+#  stage: ${opt:stage, self:provider.stage}
+  wsgi:
+    app: handler.application
+    packRequirements: false
+  settings:
+    # static_url: ${cf:cmcs-eregs-static-assets-dev.StaticURL}/
```
Comment (peggles2) [inline solution/serverless-test-files/serverless-backend.yml:38]: do we want to cleanup the commented out parts

File: solution/serverless-test-files/serverless-backend.yml [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    DB_NAME: eregs
+    USERNAME: eregsuser
+    ALLOWED_HOST: '.amazonaws.com'
+  cloudfrontInvalidate:
+#    - distributionId: ${cf:cmcs-eregs-static-assets-dev.CloudFrontDistributionId}
```
Comment (peggles2) [inline solution/serverless-test-files/serverless-backend.yml:43]: is this needed?

File: solution/serverless-test-files/serverless-backend.yml [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+#    layers:
+#      - ${cf:cmcs-eregs-static-assets-dev.PythonRequirementsLambdaLayerQualifiedArn}
+    timeout: 300
+#  populate_content:
+#   environment:
+#      SUPPLEMENTAL_CONTENT_PATH: /var/task/guidance
+#    handler: populate_content.handler
+#    layers:
+#      - ${cf:cmcs-eregs-static-assets-dev.PythonRequirementsLambdaLayerQualifiedArn}
+#    timeout: 300
```
Comment (peggles2) [inline solution/serverless-test-files/serverless-backend.yml:75]: remove all the comments that are not related to our test serverless file.

**Review by peggles2 — CHANGES_REQUESTED** [review]
Looks good but, I am getting errors for those subnetId's

File: solution/serverless-test-files/serverless-backend.yml (hunk 2 above)
Comment (thwalker6) [reply solution/serverless-test-files/serverless-backend.yml:38]: I left them in because I felt it was helpful to see what we do not include as a part of this but I'm fine with removing them

Comment (thwalker6) [reply solution/serverless-test-files/serverless-backend.yml]: removed

File: solution/serverless-test-files/serverless-backend.yml (hunk 1 above)
Comment (thwalker6) [reply solution/serverless-test-files/serverless-backend.yml]: changed it to create the subnets as opposed to just trying to call existing one.

File: solution/serverless-test-files/serverless-backend.yml [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+        #                - Ref: 'AWS::AccountId'
+        #                - 'log-group:/aws/lambda/*:*:*'
+        #        -  Effect: "Allow"
+        #           Action:
+        #             - "s3:PutObject"
+        #           Resource:
+        #             Fn::Join:
+        #               - ""
+        #               - - "arn:aws:s3:::"
+        #                 - "Ref" : "ServerlessDeploymentBucket"
```
Comment (peggles2) [inline solution/serverless-test-files/serverless-backend.yml]: are these needed too?

File: solution/serverless-test-files/serverless-backend.yml [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+        VpcId: 'vpc-079fb0427309a063e'
+        Tags:
+          - Key: "Name"
+            Value: "ServerlessSecurityGroup"
+
+    DBSecurityGroup:
+      Type: AWS::EC2::SecurityGroup
+      Properties:
+        GroupDescription: SecurityGroup for Database
+        VpcId: 'vpc-079fb0427309a063e'
```
Comment (peggles2) [inline solution/serverless-test-files/serverless-backend.yml]: were these VpcId's created in aws console manually? If so the document should reflect that a vpc needs to be created and that this file has to be updated with that VpcId.

**Review by peggles2 — CHANGES_REQUESTED** [review]

**Review by peggles2 — APPROVED** [review]
LGTM 👍
